src/backend/executor/execMain.c

```diff
@@ -296,10 +339,15 @@ CopyDirectDispatchFromPlanToSliceTable(PlannedStmt *stmt, EState *estate)
 void
 ExecutorStart(QueryDesc *queryDesc, int eflags)
 {
+	int saved_command_id = 0;
+	UPDATE_COMMAND_ID_AT_START(queryDesc, &saved_command_id);
+
 	if (ExecutorStart_hook)
 		(*ExecutorStart_hook) (queryDesc, eflags);
 	else
 		standard_ExecutorStart(queryDesc, eflags);
+
+	RESTORE_COMMAND_ID(queryDesc, saved_command_id);
```

Comment:
Do we need to restore `queryCommandId` in case of exception?

Reply:
I think no. If there are any additional logs after exception but before recovery, I think it is better for them to have cmd counter of the command that caused the failure.

Reply:
Could you add tests for such cases?

Reply:
Why don't we need to do counter restore in other places on errors, as is now done in `ExecutorRun`?

Reply:
Ok, let's do so.